swagger.yaml

```diff
@@ -0,0 +1,96 @@
+openapi: 3.0.3
+info:
+  title: Agartex Service
+  description: |-
+    This is a server for compiling Latex documents and authenticating users
+  version: 0.0.1
+servers:
+  - url: http://localhost:3000
+  - url: some_domain
+tags:
+  - name: user
+    description: Operations about user
+paths:
+  /user:
+    post:
+      tags:
+        - user
+      summary: Register user
+      description: This will register a new user with the given email and password.
+      operationId: createUser
+      requestBody:
+        description: Created user object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        200:
+          description: successful operation
+        400:
+          description: validation error
```

Comment:
by validation error you mean incorrect request body?

Reply:
This is mostly a placeholder, but yes this is just a bad request. This could be wrong format or possibly something like too weak password or invalid email

Reply:
We could make badly formatted email/password return 422 instead of 400 with some formatted information regarding the problem.

Reply:
Let's change it to 422

Reply:
_422 says "Your request is fine, it's not malformed, we understand it but for some reason we cannot process it and we don't plan to in the future"_ RFC says it means that the content is semantically incorrect, which I think fits things like a weak password, because we understand it, but it is semantically incorrect. https://httpwg.org/specs/rfc9110.html#status.422